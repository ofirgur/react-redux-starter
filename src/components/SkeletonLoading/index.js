import React from 'react';
import PropTypes from 'prop-types';

import SkeletonLoadingContainer from './style';

const SkeletonLoading = ({ width, height, circle }) => {
    return (
        <SkeletonLoadingContainer className="skeleton-loading" 
            width={width}
            height={height}
            circle={circle}
        />
    );
};

SkeletonLoading.propTypes = {
        width: PropTypes.string,
        height: PropTypes.string,
        circle: PropTypes.bool
};

export default SkeletonLoading;
