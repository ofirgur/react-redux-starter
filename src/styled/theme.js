export default {
    typography: {
        fontFamily: 'Roboto", "Helvetica", "Arial", sans-serif',
        fontSize: '16px',
        mobileFontSize: '10px'
    },
    breakpoint: {
        xs: '0px',
        sm: '600px',
        md: '960px', 
        lg: '1280px',
        xl: '1920px'
    },
    zIndex: {

    },
    direction: {
        ltr: 'ltr',
        rtl: 'rtl'
    },
    background: {
        body: '#263238',
        main: '#607d8b',
        mainHover: '#78909C'
    },
    color: {
        main: '#fff',
        purple: '#651fff',
        red: '#ff7043',
        link: '#00695C',
        hoveredLink: '#004D40'
    },
    padding: {
        main: '16px',
        mobile: '6px'
    },
    margin: {
        main: '16px',
    },
    shadow: {
        main: '0px 6px 7px -4px rgba(0,0,0,0.2),0px 11px 15px 1px rgba(0,0,0,0.14),0px 4px 20px 3px rgba(0,0,0,0.12)'
    }
};