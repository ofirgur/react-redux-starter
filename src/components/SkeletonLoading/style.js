import styled, { css, keyframes } from 'styled-components';

const SkeletonAnimation = keyframes`
    0% {
        background-position: -490px 0;
    }
    100% {
        background-position: 490px 0;
    }
`;

export default styled.div`
    width: ${props => props.width || 'auto'};
    height: ${props => props.height || '14px'};
    animation: ${SkeletonAnimation} 1s linear infinite forwards;
    animation-name: ${SkeletonAnimation};
    background: #e4e7ed;
    border-radius: 4px;
    background-image: -webkit-linear-gradient(
        left,
        #e4e7ed 0%,
        #c3cad9 20%,
        #e4e7ed 40%,
        #e4e7ed 100%
    );
    background-size: 980px 898px;
    position: relative;

    ${props =>
        props.circle &&
        css`
            border-radius: 100%;
        `};
`;
