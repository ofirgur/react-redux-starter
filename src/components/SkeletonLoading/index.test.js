import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import SkeletonLoading from './index';
import SkeletonLoadingContainer from './style';
import theme from 'styled/theme';

describe('SkeletonLoading component', () => {

    const props = {};
    const wrapper = shallow(<SkeletonLoading {...props} />);  
    
    it('it should mock the context and render correctly', () => {    
        expect(toJson(wrapper)).toMatchSnapshot();
    });

    it('it should styled correctly', () => {
        const tree = renderer.create(<SkeletonLoadingContainer theme={theme} />).toJSON();
        expect(tree).toMatchSnapshot();
    });

});