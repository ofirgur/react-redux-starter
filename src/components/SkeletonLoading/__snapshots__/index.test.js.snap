// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkeletonLoading component it should mock the context and render correctly 1`] = `
<styled.div
  className="skeleton-loading"
/>
`;

exports[`SkeletonLoading component it should styled correctly 1`] = `
.c0 {
  width: auto;
  height: 14px;
  -webkit-animation: ZQpsg 1s linear infinite forwards;
  animation: ZQpsg 1s linear infinite forwards;
  -webkit-animation-name: ZQpsg;
  animation-name: ZQpsg;
  background: #e4e7ed;
  border-radius: 4px;
  background-image: -webkit-linear-gradient( left, #e4e7ed 0%, #c3cad9 20%, #e4e7ed 40%, #e4e7ed 100% );
  background-size: 980px 898px;
  position: relative;
}

<div
  className="c0"
/>
`;
